routes/index.js

```diff
@@ -442,14 +442,20 @@ router.get('/getEvents', function(req, res) {
     collection.find({}, function(e,docs){
       var temp = new Array;
       docs.forEach(function(val){
-        if(req.session.usr.role == "EVI" && val.toEVI == "true") {
-          temp.push(val);
-        }
-        else if (req.session.usr.role == "innostaja" && val.toInnostaja == "true") {
-          temp.push(val);
-        }
-        else if (req.session.usr.role == "admin") {
-          temp.push(val);
+        if(req.session.usr.role == "EVI" && val.toEVI == "true"
+        || req.session.usr.role == "innostaja" && val.toInnostaja == "true"
+        || req.session.usr.role == "admin") {
+          if (!val.releaseTime || req.session.usr.role == "admin") { // has no release time
+            temp.push(val);
+          }
+          else { // has a release time
+            var current = new Date();
+            var releaseTime = new Date(val.releaseTime);
+            if(current > releaseTime){
+              temp.push();
+              collection.update({ _id: val._id }, {$set: { releaseTime: null }});
```

Comment:
Is it necessary to null the variable? I think it could be good to remain in object if someone would want to later check when event was released.